ios/LayoutReanimation/REAAnimationsManager.m

```diff
@@ -288,73 +242,131 @@ - (NSDictionary *)prepareDataForLayoutAnimatingWorklet:(NSMutableDictionary *)cu
   return preparedData;
 }
 
-- (void)onViewRemoval:(UIView *)view before:(REASnapshot *)before
+- (BOOL)wantsHandleRemovalOfView:(UIView *)view
 {
-  NSNumber *tag = view.reactTag;
-  ViewState state = [_states[tag] intValue];
-  if (state == Disappearing || state == ToRemove || tag == nil) {
-    return;
+  return REANodeFind(view, ^(id<RCTComponent> view) {
+    return [self->_ancestorsOfExitingViews objectForKey:view.reactTag] != nil ||
+        self->_hasAnimationForTag(view.reactTag, @"exiting");
```

Comment:
For (non-mutable) `NSString` literals, Objective-C runtime is supposed to keep a cache of literals and reuse them instead of allocating, so it shouldn't make a difference here.